packages/common/src/models/Analytics.ts

```diff
@@ -1675,6 +1691,28 @@ type BuyUSDCAddFundsManually = {
   eventName: Name.BUY_USDC_ADD_FUNDS_MANUALLY
 }
 
+// Buy Crypto
+
+type BuyCryptoEvent = {
+  eventName:
+    | Name.BUY_CRYPTO_STARTED
+    | Name.BUY_CRYPTO_ON_RAMP_OPENED
+    | Name.BUY_CRYPTO_ON_RAMP_SUCCESS
+    | Name.BUY_CRYPTO_ON_RAMP_FAILURE
+    | Name.BUY_CRYPTO_ON_RAMP_CANCELED
+    | Name.BUY_CRYPTO_ON_RAMP_CONFIRMED
+    | Name.BUY_CRYPTO_SUCCESS
```

Comment:
Do we not want events for the swap steps? And if so, wondering if we should log any of the parameters we see during that step such as quote amounts etc, if they would help us in tracing down common state during failures.

Reply:
`ON_RAMP_CONFIRMED` fires when the balance updates after the on ramp, and `BUY_CRYPTO_SUCCESS` fires after the swap succeeds. Felt redundant to have `BUY_CRYPTO_SWAP_SUCCESS` as it would fire at the same time as `BUY_CRYPTO_SUCCESS`. Thoughts?

Reply:
Actually I agree, adding analytics to quotes would be good - added events